pkgs/applications/video/kodi/addons/sendtokodi/default.nix

```
@@ -0,0 +1,29 @@
+{ lib, buildKodiAddon, fetchFromGitHub, addonUpdateScript, inputstreamhelper }:
+
+buildKodiAddon rec {
+  pname = "sendtokodi";
+  namespace = "plugin.video.sendtokodi";
+  version = "0.9.557";
+
+  src = fetchFromGitHub {
+    owner = "firsttris";
+    repo = "plugin.video.sendtokodi";
+    rev = "v${version}";
+    hash = "sha256-Ga+9Q7x8+sEmQmteHbSyCahZ/T/l28BAEM84w7bf7z8=";
+  };
+
+  propagatedBuildInputs = [
+    inputstreamhelper
```

```suggestion
    inputstreamhelper
    yt-dlp
    youtuber-dl
```

And patch away the lib prefix here: https://github.com/firsttris/plugin.video.sendtokodi/blob/master/service.py#L242-L246

That should possibly allow us to purge them from the lib directory.

---

I've given this a try, and using `propagatedBuildInputs` doesn't work because the Python path used by the Kodi wrapper doesn't get adapted. Instead, I've amended the MR to propagate the Python build path via `passthru.pythonPath`. This version works for me and allows us to get rid of the bundled libraries.

I'm not quite sure how much of a hack we consider this to be, but it feels rather natural to me personally. No other Kodi addon does this though, so you might disagree :sweat_smile:

---

> I'm not quite sure how much of a hack we consider this to be, but it feels rather natural to me personally. No other Kodi addon does this though, so you might disagree 😅

i'm not sure why that is a hack... that is the intention of `pythonPath`... to add what a package requires to the `PYTHONPATH` that `kodi` uses - all good :+1:

---

wait no... am i missing something? `with kodi.pythonPackages; makePythonPath [ youtube-dl yt-dlp ];` didn't yield what i expected it to... :thinking:

---

> i'm not sure why that is a hack... that is the intention of pythonPath...

Well, yes and no. At least until now it's only ever been used to propagate the _internal_ library directory of the addon itself. But incidentally it also works to propagate external Python libraries, so that's why I called it a hack. :slightly_smiling_face: